Python Utility Scripts

nspadaccino opened the pull request:
Summary

  This PR adds a collection of utility scripts to help with common VinylDNS Python client operations. These scripts provide ready-to-use tools for zone management and record
  manipulation tasks.

  New Scripts

  | Script                                   | Description                                                   |
  |------------------------------------------|---------------------------------------------------------------|
  | scripts/zone_dump.py                     | Outputs all records in a zone                                 |
  | scripts/search_records_by_name.py        | Outputs all records matching a record name filter             |
  | scripts/search_records_by_owner_group.py | Outputs all records owned by a certain group across all zones |
  | scripts/update_record_owner.py           | Updates ownership of a collection of records                  |

  Key Features

  - Error handling for non-existent groups and zones
  - Validation checks for old owner group before updating
  - Duplicate record filtering
  - Comprehensive logging throughout operations

  Dependencies

  - Depends on PR #78 (Added fqdn to record set options) - ✅ Already merged

nspadaccino committed:
Added new utility scripts for common python client tasks

nspadaccino committed:
update docstrings

arpit4ever commented on scripts/zone_dump.py:
```diff
@@ -0,0 +1,200 @@
+    Returns:
+        list[dict[str, str]]: List of dictionaries with keys 'fqdn', 'type', and formatted 'record_data'.
+    """
+    zone = client.get_zone_by_name(zone_name)
```
Add error handling if zone doesn't exist.

nspadaccino replied:
added

nspadaccino committed:
added check for old owner group to update_owner_group script

arpit4ever committed:
Merge branch 'master' into utility-scripts

arpit4ever commented on scripts/zone_dump.py:
```diff
@@ -0,0 +1,200 @@
+import argparse
```
Add documentation

nspadaccino replied:
I had included some general documentation in the "main" method of each script, did you want that moved, or something additional?

nspadaccino replied:
Moved documentation from main to the top of the script

JoshSEdwards committed:
Update dependencies and Python compatibility to address security vulnerabilities (#82)

* Update dependencies and Python compatibility

Change pip install to pip3 install in bootstrap script for clarity.
Remove future package with security vulnerabilities from setup.py dependencies.
Update tox.ini to support Python 3.8 to 3.12 and remove older Python versions.
Refactor client.py to eliminate usage of future library, replacing it with standard library methods.

* Re-add python 3.7 environment to tox config

* Re-add all python 3 versions to tox config

---------

Co-authored-by: Arpit Shah <arpit_shah@cable.comcast.com>

nspadaccino committed:
Merge branch 'master' into utility-scripts

arpit4ever commented on scripts/search_records_by_owner_group.py:
```diff
@@ -0,0 +1,242 @@
+            env_vars["VINYLDNS_SECRET_KEY"],
+        )
+
+        records = search_records_by_owner_group(client, args.owner_group_id, record_name_filter_list)
```
Verify if owner_group_id is a valid group id.

nspadaccino replied:
added

arpit4ever reviewed (changes requested):
Initial review comments.

nspadaccino committed:
added error handing for non existent groups and zones

arpit4ever commented on scripts/update_record_owner.py:
```diff
@@ -0,0 +1,151 @@
+        old_owner_group_id (str): The old owner group ID to update.
+        new_owner_group_id (str): The new owner group ID to apply.
+    """
+    new_group = client.get_group(new_owner_group_id)
```
Similar check for old_owner_group_id required.

nspadaccino replied:
added

arpit4ever commented on scripts/update_record_owner.py:
```diff
@@ -0,0 +1,151 @@
+
+    for record_name in record_names:
+        try:
+            response = client.search_record_sets(record_name_filter=record_name,
```
Add a check if the record doesn't exist.
Also a check if the record is not owned by old_group

nspadaccino replied:
added

arpit4ever reviewed (changes requested):
Couple of other minor check to be added. 
Otherwise all 4 script looks good.

Sokitha-21 commented on scripts/search_records_by_owner_group.py:
```diff
@@ -0,0 +1,247 @@
+            env_vars["VINYLDNS_SECRET_KEY"],
+        )
+
+        records = search_records_by_owner_group(client, args.owner_group_id, record_name_filter_list)
```
Add error handling for cases where records returns an empty list.

nspadaccino replied:
Added logging for this case.

Sokitha-21 commented on scripts/search_records_by_name.py:
```diff
@@ -0,0 +1,201 @@
+            env_vars["VINYLDNS_SECRET_KEY"],
+        )
+
+        records = search_records(client, args.record_name_filter)
```
Add error handling for cases where records returns an empty list.

nspadaccino replied:
Added logging for this case.

Sokitha-21 commented on scripts/zone_dump.py:
```diff
@@ -0,0 +1,202 @@
+            env_vars["VINYLDNS_SECRET_KEY"],
+        )
+
+        records = dump_zone_records(client, args.zone_name)
```
Add error handling for cases where records returns an empty list.

nspadaccino replied:
Added logging for this case.

Sokitha-21 reviewed (changes requested):
Other than the minor changes requested, everything LGTM.

nspadaccino committed:
updated documentation, updated conditonals and logging for update_record_owner.py

nspadaccino committed:
added additional checks for update owner group script

nspadaccino committed:
more logging

arpit4ever reviewed (approved):
LGTM

nspadaccino committed:
omit duplicate records
